### PR TITLE
1 0 es6.1

### PR DESCRIPTION
**Pull request by jettro**
Migration to the elasticsearch 6.1 release.

### CODE REVIEW

File: src/main/java/com/o19s/es/ltr/query/LtrQueryBuilder.java [hunk 1]
```diff
@@ -107,29 +114,49 @@ private static void doXArrayContent(String field, List<QueryBuilder> clauses, XC
...
         List<PrebuiltFeature> features = new ArrayList<PrebuiltFeature>(_features.size());
-        for(QueryBuilder builder: _features) {
+        for (QueryBuilder builder : _features) {
             features.add(new PrebuiltFeature(builder.queryName(), builder.toQuery(context)));
         }
         features = Collections.unmodifiableList(features);
-        // pull model out of script
-        LtrRanker ranker = (LtrRanker) context.getExecutableScript(_rankLibScript, ScriptContext.Standard.SEARCH).run();
+
+        ExecutableScript.Factory factory = context.getScriptService().compile(_rankLibScript, new ScriptContext<>(
```
[the remark this answers is not in the document]
Comment (jettro) [reply src/main/java/com/o19s/es/ltr/query/LtrQueryBuilder.java]: I am not confident about the solution for this ScriptContext yet. It took me some time to try and understand. But be happy to refactor to something better. If we want to keep it, wil try to find more information.

**Comment by jettro** [comment]
Made some improvements based on the feedback so far

**Comment by nomoa** [comment]
LGTM
Thanks for your patch this helps a lot!
I'm OK to merge unless you prefer to work on top of this PR.
I think this is sane base to address next issues:
 - decide what to do with this ScriptEngine
- fix/rework integration tests.

**Comment by jettro** [comment]
I think I am ready, I am fine with you merging the pul request.

**Comment by nomoa** [comment]
Thanks!

**Pull request merged**